backend/libraries/ledger_utils/src/lib.rs

```diff
@@ -4,9 +4,56 @@ pub mod sns;
 
 use candid::{CandidType, Principal};
 use ic_ledger_types::{AccountIdentifier, Memo, Subaccount, Timestamp, Tokens, TransferArgs, DEFAULT_SUBACCOUNT};
+use icrc_ledger_types::icrc1::account::Account;
 use serde::{Deserialize, Serialize};
 use sha256::sha256;
-use types::{CanisterId, TransactionHash};
+use types::{
+    nns::UserOrAccount, CanisterId, CompletedCryptoTransaction, Cryptocurrency, FailedCryptoTransaction,
+    PendingCryptoTransaction, TimestampNanos, TransactionHash, UserId,
+};
+
+pub fn create_pending_transaction(
+    token: Cryptocurrency,
+    amount: Tokens,
+    user_id: UserId,
+    now_nanos: TimestampNanos,
+) -> PendingCryptoTransaction {
+    let principal = Principal::from(user_id);
```

Comment:
Lifting this out seems unnecessary since it is only used in one place and in one of the branches